pkg/apis/fake/v1alpha1/db_types.go

```diff
@@ -0,0 +1,265 @@
+/*
+Copyright 2019 The KubeCarrier Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	"fmt"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// DBSpec defines the desired state of DB
+type DBSpec struct {
+	// RootPassword is root account password for this data. Leave blank for auto-generation
+	RootPassword string `json:"rootPassword"`
+
+	// DatabaseName of the created database at start up
+	DatabaseName string `json:"databaseName"`
+
+	// DatabaseUser for created database
+	DatabaseUser string `json:"databaseUser"`
+
+	// DatabasePassword for the created database. Leave blank for auto-generation
+	DatabasePassword string `json:"databasePassword"`
+
+	// E2E tests params
+	Config Config `json:"config,omitempty"`
+}
+
+// Config defines the e2e tests params
+type Config struct {
+	ReadyAfterSeconds    int `json:"readyAfterSeconds,omitempty"`
+	DeletionAfterSeconds int `json:"deletionAfterSeconds,omitempty"`
```

Comment:
What does this parameter does? Could you also add also fill the docs for these fields.

e.g. `DeletionAfterSeconds` is the deletion delay until this object is deleted by the controller.

Reply:
> What does this parameter do?
Yep, it's a timeout that the operator waits till removing the finalizer.

Reply:
The word timeout has the wrong connotation in this context (timeout is usually used to determined maximum time for an operation to take until it's deemed a failure, which here isn't the case). Could you use the word duration instead, e.g. duration after which operator will mark DB as Rady